test/gurka/test_matrix.cc

```diff
@@ -538,3 +538,123 @@ TEST(StandAlone, CostMatrixShapes) {
 
   res.erase();
 }
+
+class DateTimeTest : public ::testing::Test {
+protected:
+  // check both with and without time zones present
+  static gurka::map map;
+  static gurka::map map_tz;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 1500;
+
+    // ~ are approximate time zone crossings
+    const std::string ascii_map = R"(
+      A----------B
+      |          |
+      C          D
+      |          |
+      ~          ~
+      |          |
+      |          |
+      E          F
+      |          |
+      G----------H
+    )";
+
+    const gurka::ways ways = {{"AC", {{"highway", "residential"}}},
+                              {"CE", {{"highway", "residential"}}},
+                              {"EG", {{"highway", "residential"}}},
+                              {"GH", {{"highway", "residential"}}},
+                              {"HF", {{"highway", "residential"}}},
+                              {"FD", {{"highway", "residential"}}},
+                              {"DB", {{"highway", "residential"}}},
+                              {"BA", {{"highway", "residential"}}}};
+
+    const auto layout = gurka::detail::map_to_coordinates(ascii_map, gridsize, {-8.5755, 42.1079});
+    map = gurka::buildtiles(layout, ways, {}, {}, "test/data/time_zone_matrix_no_tz");
+    map_tz = gurka::buildtiles(layout, ways, {}, {}, "test/data/time_zone_matrix",
+                               {{"mjolnir.timezone", "test/data/tz.sqlite"},
+                                {"thor.source_to_target_algorithm", "costmatrix"},
+                                {"service_limits.max_timedep_distance_matrix", "50000"}});
+  }
+};
+gurka::map DateTimeTest::map = {};
+gurka::map DateTimeTest::map_tz = {};
+
+TEST_F(DateTimeTest, DepartAt) {
+  rapidjson::Document res_doc;
+  std::string res;
+  {
+    auto api = gurka::do_action(valhalla::Options::sources_to_targets, map_tz, {"A", "G"}, {"A", "G"},
+                                "auto",
+                                {{"/prioritize_bidirectional", "true"},
```

Comment:
We're only testing `costmatrix` now because of a bug discovered in `timedistancematrix` that causes the timezone info of the last found source to be set on all source locations.

Reply:
I like the tests. It's a lot but it's very easy to follow and we won't get failures in some opaque functions